engine/samples/cars/main.cpp

```diff
@@ -17,30 +17,47 @@ using cubos::core::gl::Grid;
 using cubos::core::gl::Palette;
 using namespace cubos::engine;
 
+/// [Get handles to assets]
 static const Asset<Grid> CarAsset = AnyAsset("059c16e7-a439-44c7-9bdc-6e069dba0c75");
 static const Asset<Palette> PaletteAsset = AnyAsset("1aa5e234-28cb-4386-99b4-39386b0fc215");
+/// [Get handles to assets]
 
+/// [Spawner resource]
 struct Spawner
 {
     float timer = 0.0F;
     int x = -1;
     int y = -1;
 };
+/// [Spawner resource]
 
-static void settings(Write<Settings> settings)
+/// [Set settings]
+static void settingsSystem(Write<Settings> settings)
 {
     settings->setString("assets.io.path", SAMPLE_ASSETS_FOLDER);
 }
+/// [Set settings]
 
-static void setup(Commands cmds, Write<Assets> assets, Write<Renderer> renderer, Write<ActiveCameras> activeCameras)
+/// [Load, add materials and set pallete]
+static void setPalleteSystem(Write<Assets> assets, Write<Renderer> renderer)
 {
     // Load the palette asset and add two colors to it.
     auto palette = assets->write(PaletteAsset);
-    auto black = palette->add({{0.1F, 0.1F, 0.1F, 1.0F}});
-    auto white = palette->add({{0.9F, 0.9F, 0.9F, 1.0F}});
+    palette->add({{0.1F, 0.1F, 0.1F, 1.0F}});
+    palette->add({{0.9F, 0.9F, 0.9F, 1.0F}});
 
     // Set the renderer's palette to the one we just modified.
     (*renderer)->setPalette(*palette);
+}
```

Comment:
Perhaps on this case its best to keep this system merged with `spawnFloorSystem`. You end up having less repetition and the code is less confusing, since we don't need to search for the materials again.